collector/lib/EnvVar.h

```diff
@@ -46,6 +48,54 @@ class EnvVar {
   mutable std::once_flag init_once_;
 };
 
+template <typename T, typename ParseT>
+class OptionalEnvVar {
```

Comment:
This is a pretty Rusty class if ever I've seen one, and it's great :smile: :crab: